extensions/wikia/CommunityPage/tests/LinkHelperTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+require_once( $IP . '/extensions/wikia/CommunityPage/helpers/LinkHelper.php' );
+
+class LinkHelperTest extends WikiaBaseTest {
+	const ANON = false;
+	const LOGGED_IN = true;
+	const ANNON_EDITS_ALLOWED = false;
+	const ANNOON_EDITS_FORBIDDEN = true;
+
+	const ARTICLE_NAME = 'test_article';
+	const ARTICLE_LOCAL_URL = '/wiki/test_article';
+	const ARTICLE_LOCAL_EDIT_URL = '/wiki/test_article?veaction=edit';
+	const SIGNUP_URL_WITH_EDIT = '/wiki/Special:SignUp?returnto=test_article&returntoquery=veaction%253Dedit&type=login';
+	const SIGNUP_URL_WITHOUT_EDIT = '/wiki/Special:SignUp?returnto=test_article&type=login';
+
+	/**
+	 * @dataProvider forceLoginLinkTestCases
+	 */
+	public function testForceLoginLink( $user, $editMode, $disableAnonEdits, $expectedLink ) {
+		$mockTitle = $this->getMock( 'Title', [ 'getEscapedText', 'getLocalURL' ] );
```

Comment:
You don't have to mock it, just use:
`$title = Title::newFromText('test_article');`